// 定义公用message，注意与ComDefine.proto区别

import "ProtoBuffer/Com.proto";
package ProtoBuf;

//请求移动  CLIENT_MOVE_REQ
message MoveReq
{
	required Vector3PB pos = 1;	  //生物当前位置
	required Vector2PB speed = 2; //生物移动速度(速度向量),为 0 时表示停止移动
	required Vector2PB dir = 3;	  //生物移动的方向
	optional int32 move_type = 4; //移动类型 1-技能技能，其他类型后面用到的时候再定义
}


//生物移动广播 MOVE_BROAD_RSP
message MoveBroadRsp
{
	required uint64 cid = 1;		//生物cid
	required Vector3PB pos = 2;		//生物当前位置
	required Vector2PB speed = 3;	//生物移动速度(速度向量)
	required Vector2PB dir = 4;		//生物移动的方向
}


//请求瞬移 MOVE_TELEPORT_REQ = 604
message MoveTeleportReq
{
	/*
		type：瞬移类型，详见  MoveTeleportRsp 消息中 Type 枚举

		任务  type = 6, param1:动态任务ID， param2: point.xlsx->point 表配置ID
	*/

	required int32 type = 1;   //瞬移类型	
	optional int64 param1 = 2; //类型参数1
	optional int64 param2 = 3; //类型参数2
}

//返回瞬移结果 MOVE_TELEPORT_RESULT_RSP
message MoveTeleportResultRsp
{
	required int32 ret = 1;		//返回码
	optional int32 type = 2;    //瞬移类型	
	optional int64 param1 = 3;  //类型参数1
	optional int64 param2 = 4;  //类型参数2
}


//瞬移消息 MOVE_TELEPORT_RSP
message MoveTeleportRsp
{
	//瞬移类型
	enum Type
	{
		common   =1;	//普通瞬移
		skill	 =2;	//技能传送
		dup		= 3;	//副本内瞬移
		relive	= 4;	//复活瞬移
		trans	= 5;    //帮派传功
		task	= 6;    //任务
	}

	required int32  type	   = 1;	//瞬移类型
	required uint64 cid		   = 2;	//瞬移的生物cid
	required Vector3PB pos     = 3;	//瞬移的目标坐标
}
